src/engine/source/router/src/runtimePolicy.cpp

```diff
@@ -32,14 +32,144 @@ std::optional<base::Error> RuntimePolicy::build(std::shared_ptr<builder::Builder
 
 std::optional<base::Error> RuntimePolicy::processEvent(base::Event event)
 {
-    if (!m_controller)
+    if (!m_spController)
     {
         return base::Error {fmt::format("Policy '{}' is not built", m_asset)};
     }
     auto result = base::result::makeSuccess(event);
-    m_controller->ingestEvent(std::make_shared<base::result::Result<base::Event>>(std::move(result)));
+    m_spController->ingestEvent(std::make_shared<base::result::Result<base::Event>>(std::move(result)));
 
     return std::nullopt;
 }
 
+void RuntimePolicy::subscribeToOutput()
+{
+    auto subscriber = rxcpp::make_subscriber<rxbk::RxEvent>(
+        [&](const rxbk::RxEvent& event)
+        {
+            std::stringstream output;
+            output << event->payload()->prettyStr() << std::endl;
+            auto result = m_output.emplace(m_asset, output.str());
+            if (!result.second)
+            {
+                result.first->second = output.str();
+            }
+        });
+
+    m_spController->getOutput().subscribe(subscriber);
+}
+
+void RuntimePolicy::listenAllTrace()
+{
+    m_spController->listenOnAllTrace(rxcpp::make_subscriber<std::string>(
+        [&](const std::string& trace)
+        {
+            const std::string opPatternTrace = R"(\[([^\]]+)\] \[condition\]:(.+))";
```

Comment:
Can we define a `constexpr` for this?

Reply:
it can't be done. error: expression must have a constant value